include/openenclave/bits/socket.h

```diff
@@ -35,7 +35,7 @@ typedef struct oe_provider_fd_set {
 
 typedef uint16_t oe_sa_family_t;
 
-typedef int oe_socklen_t;
+typedef size_t oe_socklen_t;
```

Comment:
This change is right for POSIX but wrong for Winsock.   We may want an ifdef, in order to allow porting code from Winsock apps.

Reply:
I've reverted all the changes I made to sockets.

Originally, I wanted to clean up warnings, but this will require some thinking. For instance, we can't use `oe_socklen_t` in an EDL file because the type is defined in this header, but you can't include this header without including `enclave.h` in the EDL. I don't know what the effects of that would be, if any. Additionally, we need to make sure that all types match in size when consumers are compiled for x64 (I've seen warnings to that effect).

Reply:
If it helps, it's ok for the EDL to use a 64-bit type even on 32-bit platforms.